test/middleware/FileRestrictionsMiddleware.test.ts

```diff
@@ -0,0 +1,281 @@
+
+    const DEFAULT_ATTACHMENTS = createDefaultAttachments();
+
+    const getRequestSubsitute = (
+        params: any,
+        admin: boolean,
+        data?: Partial<ApplicationData>,
+        permissions?: any): Request => {
+
+        const session = createSession('secret', true, admin, permissions);
+        session.data[SessionKey.ExtraData] = {
+            appeals: { ...data } || undefined
+        };
+
+        session.data[SessionKey.SignInInfo]![SignInInfoKeys.UserProfile]!.id = userId;
+
+        return {
+            session: Maybe.of(session),
+            params
+        } as Request;
+    };
+
+    const isAdmin = true;
+
+    describe('As a CH Internal User session', () => {
+
+        const appData: Partial<ApplicationData> = { appeal: createDefaultAppeal(DEFAULT_ATTACHMENTS) };
+        const fileRestrictionsMiddleware = new FileRestrictionsMiddleware();
+
+        it('should call next if the correct permissions are present in the session', () => {
+
+            const request = getRequestSubsitute(
+                { fileId: DEFAULT_ATTACHMENTS[0].id },
+                isAdmin,
+                appData,
+                { [AppealsPermissionKeys.download]: 1, [AppealsPermissionKeys.view]: 1 }
+            );
+
+            const response = createSubstituteOf<Response>();
+            const next = createSubstituteOf<NextFunction>();
+
+            fileRestrictionsMiddleware.handler(request, response, next);
+            next.received();
+            response.didNotReceive();
+
+        });
+
+        it('should redirect to an error page if user does not have the correct permissions', () => {
+
+            const requestNoPermissions = getRequestSubsitute({ fileId: DEFAULT_ATTACHMENTS[0].id }, isAdmin, appData);
+            console.log(requestNoPermissions.session.__value.data.signin_info?.user_profile);
+            const requestInvalidPermissions = getRequestSubsitute(
+                { fileId: DEFAULT_ATTACHMENTS[0].id },
+                true,
+                appData,
+                { '/some-permission/not-appeals': 1 }
+            );
+
+            let response = createSubstituteOf<Response>();
+            let next = createSubstituteOf<NextFunction>();
+
+            fileRestrictionsMiddleware.handler(requestNoPermissions, response, next);
+            response.received().status(FORBIDDEN);
+            response.received().render('error-custom', {
+                heading: 'You are not authorised to download this document'
+            });
+            next.didNotReceive();
+
+            response = createSubstituteOf<Response>();
+            next = createSubstituteOf<NextFunction>();
+
+            fileRestrictionsMiddleware.handler(requestInvalidPermissions, response, next);
+            response.received().status(FORBIDDEN);
+            response.received().render('error-custom', {
+                heading: 'You are not authorised to download this document'
+            });
+            next.didNotReceive();
+
```

Comment:
Empty line!

Reply:
I'll fix the formatting on all the files :)